R/deleteGates.R

```diff
@@ -0,0 +1,39 @@
+#' Delete gate
+#'
+#' Deletes a gate or a tailored gate family
+#' Compound gates (quadrant and split) are made up of "sectors." 
+#' Quadrant gates have four sectors (upper-right, upper-left, lower-left, lower-right) and split gates have two sectors (left and right).
+#' In addition to the top-level GID (like simple gates), these gates have model.gids and names lists that specify the GID and name for each sector, in the order shown above. 
+#' Populations using compound gates must reference these sector GIDs; referencing the top-level GID of a compound gate is meaningless.
+#' 
+#'
+#' @param experimentId ID of experiment.
+#' @param gid ID of gate family.
+#' @param gateId ID of gate.
+#' @param exclude Gate ID to exclude from deletion.
+#' @export
+#' @examples
+#' \dontrun{
+#' deleteGate(experimentId, gateId)
+#' }
+#' 
+#' TODO:(ge) This needs to account for compound gates (Quadrant, Split)
+
+deleteGates = function(experimentId, gid, gateId = NULL, exclude = NULL) {
+  checkDefined(experimentId)
+  experimentId = lookupByName("experiments", experimentId)
+  #checkDefined(gateId)
+  #gateId = lookupByName(paste("experiments", experimentId, "gates", sep = "/"), gateId, "filename")
+  
+  if (!is.null(gateId)){
+    baseDelete(paste("experiments", experimentId, "gates", gateId, sep = "/"))
+  } else if (is.null(exclude)){
+    base = paste("experiments", experimentId, "gates", sep = "/")
+    url = sprintf("%s?gid=%s", base, gid)
+    baseDelete(url)
+  } else{
+    base = paste("experiments", experimentId, "gates", sep = "/")
+    url = sprintf("%s?gid=%s&exclude=%s", base, gid, exclude)
+    baseDelete(url)
+  }
+}
```

Comment:
nit: can you add a newline at the end of the file please? (Prob an editor setting will do this automatically.)